meson.build

```diff
@@ -0,0 +1,281 @@
+project(
+  'exiv2',
+  'cpp',
+  version: '1.0.0',
+  meson_version: '>=0.49.0',
+  default_options: ['warning_level=1', 'cpp_std=c++17'],
+)
+
+cpp = meson.get_compiler('cpp')
+if host_machine.system() == 'windows' and get_option('default_library') != 'static'
+  dllapi = '-DEXIV2API=__declspec(dllexport)'
+elif cpp.has_function_attribute('visibility')
+  dllapi = '-DEXIV2API=__attribute__((visibility("default")))'
+else
+  dllapi = '-DEXIV2API='
+endif
+
+exiv_api = configuration_data()
+exiv_conf = configure_file(output: 'exiv2lib_export.h', configuration: exiv_api)
+
+cdata = configuration_data()
+cdata.set('EXV_PACKAGE_NAME', meson.project_name())
+ver = meson.project_version().split('.')
+cdata.set('PROJECT_VERSION_MAJOR', ver[0])
+cdata.set('PROJECT_VERSION_MINOR', ver[1])
+cdata.set('PROJECT_VERSION_PATCH', ver[2])
+cdata.set('PROJECT_VERSION_TWEAK', 9)
+cdata.set('PROJECT_VERSION', '@0@.@1@'.format(meson.project_version(), cdata.get('PROJECT_VERSION_TWEAK')))
+cdata.set('EXV_PACKAGE_VERSION', '@0@.@1@'.format(meson.project_version(), cdata.get('PROJECT_VERSION_TWEAK')))
+cdata.set('EXV_PACKAGE_STRING', '@0@ @1@'.format(meson.project_name(), cdata.get('PROJECT_VERSION')))
+
+cdata.set('EXV_HAVE_LIBPROC_H', cpp.has_header('libproc.h'))
+cdata.set('EXV_HAVE_SYS_MMAN_H', cpp.has_header('sys/mman.h'))
+cdata.set('EXV_HAVE_UNISTD_H', cpp.has_header('unistd.h'))
+
+cdata.set('EXV_HAVE_MMAP', cpp.has_function('mmap'))
+cdata.set('EXV_HAVE_MUNMAP', cpp.has_function('munmap'))
+cdata.set('EXV_HAVE_STRERROR_R', cpp.has_function('strerror_r'))
+cdata.set('EXV_STRERROR_R_CHAR_P', not cpp.compiles('#include <string.h>\n int strerror_r(int, char*, size_t); int main(){}'))
+
+cdata.set('EXV_ENABLE_BMFF', get_option('bmff'))
+cdata.set('EXV_HAVE_LENSDATA', get_option('lensdata'))
+cdata.set('EXV_ENABLE_VIDEO', get_option('video'))
+
+deps = []
+deps += cpp.find_library('ws2_32', required: host_machine.system() == 'windows')
+deps += dependency('INIReader')
+brotli_dep = dependency('libbrotlidec', disabler: true, required: false)
+if brotli_dep.found()
+  deps += brotli_dep
+endif
+
+curl_dep = dependency('libcurl', disabler: true, required: get_option('curl'))
+if curl_dep.found()
+  deps += curl_dep
+endif
+
+zlib_dep = dependency('zlib', disabler: true, required: get_option('png'))
+if zlib_dep.found()
+  deps += zlib_dep
+endif
```

Comment:
As a matter of curiosity, why do these need to be `disabler: true`?

It only takes effect if the `.found()` method returns false, but everywhere you seem to first check for that anyway.

Aside: if you do *not* use disablers, then it is safe to append them unconditionally to `deps +=`. Not-found dependencies are no-ops by default and possess no compiler flags or linkable libraries, so they just do nothing when added... (that's why you also need to explicitly check `.found()` and set `EXV_HAVE_*`).

Reply:
It’s mostly a hack to get the later set calls to not work and have the cmake and meson config files be more sinmilar.

Reply:
The later `set()` calls are "working" and produce `#undef`.

Reply:
the difference is

```
> /* #undef EXV_HAVE_LIBINTL_H */
```

vs

```
> #undef EXV_HAVE_LIBINTL_H
```